### PR TITLE
fix(16546) - Adapters create metrics in a consistent namespace

### PR DESCRIPTION
**Pull request by simon622**
The behaviour of metrics was non-deterministic, and thus using any sort of schema to retreive different types was impossible. This PR couples configuration to adapter type instances more tightly using generics and wraps the metrics generation and exposes a helper for the adapter instances to consume this service consistently.

### CODE REVIEW

File: hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/metrics/ProtocolAdapterMetricsHelper.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    /**
+     * Increment an arbitrary counter in the adapter instance namespace
+     * @param metricName - the metric name to be incremented (inside) the adapter namespace
+     */
+    public void increment(final @NotNull String metricName){
+        Preconditions.checkNotNull(metricName);
+        metricRegistry.counter(createAdapterMetricsNamespace(metricName)).inc();
+    }
+
+    protected String createAdapterMetricsNamespace(String suffix){
```
Comment (sfrehse) [inline hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/metrics/ProtocolAdapterMetricsHelper.java:73]: add @NotNull

**Review by sfrehse — APPROVED** [review]
There are tests missing for this. We continuously reduce the coverage.

**Comment by simon622** [comment]
Added functional tests (IT tests exist in the test repository) - we need a strategy for this because the PR doesnt pull them into master - this is not the first time this has been an issue.

File: hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/metrics/ProtocolAdapterMetricsHelper.java (hunk 1 above)
Comment (simon622) [reply hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/metrics/ProtocolAdapterMetricsHelper.java]: was updated in subsequent push

**Pull request merged**